### PR TITLE
Add delete for order shipments

### PR DESCRIPTION
**Pull request by maxheyn**

### CODE REVIEW

File: bigc/resources/orders.py [hunk 1]
```diff
@@ -88,3 +88,8 @@ def create_shipment(self, order_id: int, *, order_address_id: int, items: list[d
         }
 
         return self._api.v2.post(f'/orders/{order_id}/shipments', json=payload)
+
+    def delete_shipment(self, order_id: int, shipment_id: int) -> dict:
```
Comment (maxheyn) [inline bigc/resources/orders.py:92]:
```suggestion
    def delete_shipment(self, order_id: int, shipment_id: int) -> None:
```

File: bigc/resources/orders.py [hunk 2]
```diff
@@ -88,3 +88,8 @@ def create_shipment(self, order_id: int, *, order_address_id: int, items: list[d
         }
 
         return self._api.v2.post(f'/orders/{order_id}/shipments', json=payload)
+
+    def delete_shipment(self, order_id: int, shipment_id: int) -> None:
+        """Deletes the specified order shipment for the specified order by their IDs"""
+
+        return self._api.v2.delete(f'/orders/{order_id}/shipments/{shipment_id}')
```
Comment (r-thomson) [inline bigc/resources/orders.py:95]:
```suggestion
        self._api.v2.delete(f'/orders/{order_id}/shipments/{shipment_id}')
```

**Review by r-thomson — CHANGES_REQUESTED** [review]

**Review by r-thomson — APPROVED** [review]

**Review by Iapetus-11 — CHANGES_REQUESTED** [review]
What if we wrapped the perform_create for a shipment in an atomic block and then tried saving the Shipment to our database before creating the shipment in bigcommerce, therefore removing the need for the delete_shipment method.

**Comment by maxheyn** [comment]
We don't need this anymore but if we ever need this in the future then here it is.

**Pull request closed without merging**